### PR TITLE
Add NoSuitableDataProviderError

### PR DESCRIPTION
**Pull request by epa095**
To indicate that we have requested an tag which no dataproviders are
able to handle

### CODE REVIEW

File: gordo_components/data_provider/providers.py [hunk 1]
```diff
@@ -23,6 +23,10 @@
 logger = logging.getLogger(__name__)
 
 
+class NoSuitableDataProviderError(ValueError):
+    pass
```
Comment (milesgranger) [inline gordo_components/data_provider/providers.py:27]: Do you think we should have a common collection for gordo exceptions?

**Review by milesgranger — APPROVED** [review]
Approving with single comment, take it or leave it. But also, consider reviewing some of my PRs?

File: gordo_components/data_provider/providers.py (hunk 1 above)
Comment (epa095) [reply gordo_components/data_provider/providers.py:27]: By collection do you mean to put them somewhere together? IDK, I feel like errors like these belongs in only class which throws them, they are part of the API of this class. But more generall errors can definitely be collected somewhere.

Comment (milesgranger) [reply gordo_components/data_provider/providers.py:27]: Ya, I mean libs like `sklearn` have them in `sklearn.exceptions` and in general that seems fairly standard. But I do not have strong opinions about this, yet. :-)

**Pull request merged**